### DISCUSSION TITLE
Negative mu_SIG in the ranking

### DISCUSSION THREAD

**Discussion by jakobnov** [histfitter/histfitter#118] [Q&A] [answered]
Dear HistFitter experts,

I’m running simultaneous SR+CR fit and I get best-fit value for a mu_SIG parameter with a rather large uncertainty:

mu_SIG = 0.25 +/- 57.39

mu_SIG range is set to [0,100] in the fit config. Then I’m running ranking plot with SystRankingPlotMP.py script:

```
python $HISTFITTER/scripts/SystRankingPlotMP.py -f sl_e_WjetsCR,sl_m_WjetsCR,met_e_WjetsCR,met_m_WjetsCR,sl_e_StopCR,sl_m_StopCR,met_e_StopCR,met_m_StopCR,sl_e_dHHCR,sl_m_dHHCR,met_e_dHHCR,met_m_dHHCR,sl_e_SR,sl_m_SR,met_e_SR,met_m_SR -w results/ResolvedHHbbVVFit_NNXHSX2000S240bbWW1lep/Exclusion_NonRes_combined_NormalMeasurement_model_afterFit.root -p mu_SIG --lumi 139 --dataSet asimovData --sysOnly
```

Best-fit value for mu_SIG gets pulled to a negative value when refit by the ranking script. Apparently originally set limits get ignored in the ranking and best-fit value uncertainty is taken to set boundaries on the floating range of mu_SIG. Is there a way to enforce original bounds in the ranking script, since negative mu_SIG is unphysical?

Thanks,
Jakob

**Comment by longjon929** [comment]
Hi @jakobnov ,

Hmm, off the top of my head I'm not sure. Maybe someone else has a better idea It sounds like for some reason the range is getting lost when the refits are performed in this script with Util::FitPdf().  Maybe you can call RooRealVar::SetRange() https://root.cern.ch/doc/master/classRooRealVar.html#ad7f620f66a5516d69df2d4da645acdbc in the script around refit_parameter(opts, index) I suppose.  I'm not sure if setting const to some parameters is causing this.  HistFitter -f might be calling doFreeFit() instead.  I haven't looked at this for a while so I'm not 100% sure.

That's probably not super helpful.  I would try to print the parameter values and ranges in the script and see if the range is lost, then set it with the RooRealVar function.  Hopefully that works.

Best,
Jonathan

**Reply by jakobnov** [reply, marked as answer]
Hi @longjon929 ,

thank you for your reply! After performing some checks, I realized that delta_mu in the ranking plot is defined as mu_best_fit - mu_varied (not what I assumed mu_varied - mu_best_fit). Therefore the range of mu_SIG is fine. I had a problem of one-sided impacts, which I thought was due to minimum at negative mu_SIG value, but turns out it was due to a differece in the asimov dataset. This issue is fixed now.

Thanks,
Jakob
